### PR TITLE
support for windows paths

### PR DESCRIPTION
**Pull request by kvnkho**
Note that there is an unexpected behavior in pyfilesystem. Just using the letter of the drive returns the current directory. A slash is needed after the drive to avoid this. 

```
open_fs('C:')
OSFS('C:\Users\Kevin\Desktop\Projects')

open_fs('C:/')
OSFS('C:\')
```
This is why line 78 of fs.py looks like that. Line 79 is intended to strip both leading and training slashes from the relative path. What's nice about this solution is it should work for users with multiple drives (C:/, D:/, etc.)

I am really unsure of this condition to check if the path starts with a capital letter. I think it should work, but I am unsure about any side effects this could cause with other file systems.

There is 1 edge case where I think this solution fails. It's if a Windows user makes a path with forward slashes like:

`/temp/a/b`

It will be dealt with as a Linux path and `os.path.abspath` will insert the root drive in line 88. I am not sure if this can be dealt with.

### CODE REVIEW

File: triad/collections/fs.py [hunk 1]
```diff
@@ -67,26 +67,36 @@ def _delegate(self, path) -> Tuple[FSBase, str]:
...
-        if path is None:
-            raise ValueError("path can't be None")
+        if path is None or path == "":
+            raise ValueError("path can't be None or empty")
         path = self._modify_path(path)
-        if path.startswith("file://"):
-            path = path[6:]
-        if path.startswith("/"):
+        # Windows
+        if path[0].isupper() or "\\" in path:
```
Comment (goodwanghan) [inline triad/collections/fs.py:74]: `and`?

Comment (kvnkho) [reply triad/collections/fs.py:74]: I think _or_ because a path can be something like `\\temp\\a.csv`. It just ends up being treated as absolute.

Comment (goodwanghan) [reply triad/collections/fs.py:74]: what about `TMP/file.txt`?

**Comment by kvnkho** [comment]
Moving implementation to PureWindowsPath from pathlib

**Pull request closed without merging**